py/redrock/zfind.py

```diff
@@ -546,13 +588,20 @@ def zfind(targets, templates, mp_procs=1, nminima=3, archetypes=None, priors=Non
             del allresults[tid]['meta']
 
         # Standardize column sizes
-        if allzfit['subtype'].dtype != '<U20':
-            allzfit.replace_column('subtype', allzfit['subtype'].astype('<U20'))
+        if allzfit['subtype'].dtype != '<U32':
+            allzfit.replace_column('subtype', allzfit['subtype'].astype('<U32'))
 
         if allzfit['spectype'].dtype != '<U6':
             allzfit.replace_column('spectype',allzfit['spectype'].astype('<U6'))
 
-        maxcoeff = np.max([t.template.nbasis for t in templates])
+        if not per_camera:
+            maxcoeff = np.max([t.template.nbasis for t in templates])
+        else:
+            ncam = 3 # three cameras of DESI:: b, r, z
```

Comment:
See prior comment about avoiding hardcoding that DESI has 3 cameras.  And if we do hardcode it, we shouldn't hardcode it in 2 places.

Reply:
Yes, I agree, it is better to add ncam in rrdesi (with default value 3).

Reply:
This is now auto-derived from the input spectra files.